google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/PubSubImpl.java

```diff
@@ -289,54 +285,13 @@ private static ListTopicsRequest listTopicsRequest(PubSubOptions serviceOptions,
   }
 
   @Override
-  public String publish(String topic, Message message) {
-    return get(publishAsync(topic, message));
-  }
-
-  private static PublishRequest publishRequest(PubSubOptions serviceOptions, String topic,
-      Iterable<Message> messages) {
-    PublishRequest.Builder builder = PublishRequest.newBuilder();
-    builder.setTopic(PublisherClient.formatTopicName(serviceOptions.getProjectId(), topic));
-    builder.addAllMessages(Iterables.transform(messages, Message.TO_PB_FUNCTION));
-    return builder.build();
-  }
-
-  @Override
-  public Future<String> publishAsync(String topic, Message message) {
-    return transform(
-        rpc.publish(publishRequest(getOptions(), topic, Collections.singletonList(message))),
-        new Function<PublishResponse, String>() {
-          @Override
-          public String apply(PublishResponse publishResponse) {
-            return publishResponse.getMessageIdsList().get(0);
-          }
-        });
-  }
-
-  @Override
-  public List<String> publish(String topic, Message message, Message... messages) {
-    return publish(topic, Lists.asList(message, messages));
-  }
-
-  @Override
-  public Future<List<String>> publishAsync(String topic, Message message, Message... messages) {
-    return publishAsync(topic, Lists.asList(message, messages));
-  }
-
-  @Override
-  public List<String> publish(String topic, Iterable<Message> messages) {
-    return get(publishAsync(topic, messages));
-  }
-
-  @Override
-  public Future<List<String>> publishAsync(String topic, Iterable<Message> messages) {
-    return transform(rpc.publish(publishRequest(getOptions(), topic, messages)),
-        new Function<PublishResponse, List<String>>() {
-          @Override
-          public List<String> apply(PublishResponse publishResponse) {
-            return publishResponse.getMessageIdsList();
-          }
-        });
+  public Publisher publisher(TopicInfo topic) throws IOException {
+    // TODO(pongad): Provide a way to pass in the rest of the options.
+    String topicName =
+        PublisherClient.formatTopicName(getOptions().getProjectId(), topic.getName());
+    return Publisher.Builder.newBuilder(topicName)
+        .setCredentials(getOptions().getCredentials())
```

Comment:
Yes this will definitely be implemented. I plan to separate out the work, partly because I'm still not sure how all the options will get passed in yet. Once we figure out which settings to use where, I think the enhancement will essentially write itself.